shop/src/main/java/ro/msg/learning/shop/config/StrategyConfiguration.java

```diff
@@ -0,0 +1,30 @@
+package ro.msg.learning.shop.config;
+
+import lombok.AllArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import ro.msg.learning.shop.repository.LocationRepository;
+import ro.msg.learning.shop.repository.StockRepository;
+import ro.msg.learning.shop.stategy.MultipleLocationStrategy;
+import ro.msg.learning.shop.stategy.SingleLocationStrategy;
+import ro.msg.learning.shop.stategy.interfaces.IStrategy;
+
+@Configuration
+@AllArgsConstructor
+public class StrategyConfiguration {
+
+    private final StockRepository stockRepository;
+    private final LocationRepository locationRepository;
+
+    @Bean
+    public IStrategy selectStrategy(@Value("${strategy}") String strategy){
+        if(strategy.equals("single")){
```

Comment:
You can create an Enum for Strategy Type and replace the hardcoded text with Enum value.

Reply:
Enum added